st2common/st2common/validators/api/reactor.py

```diff
@@ -15,7 +15,11 @@
 
 import six
 
+from apscheduler.triggers.cron import CronTrigger
```

Comment:
I personally don't see this as an issue at all.

It might break some logical / code organization abstraction which doesn't provide much value here anyway.

Adding additional validation here allows us to provide a better user experience (we can throw early, it would get really messy and unfriendly if the exception is thrown inside rules engine when creating a timer) so I literally see no issues with that.

On top of that, we already perform validation here so I don't see how performing some additional validation breaks the abstraction. For a more specific validation (on top of what JSON schema provides) you need trigger-specific information and I see no issues with putting this here.

If we could do the same thing for other system triggers (perform additional validation which would allows us to catch issues early and provide a better user experience) I would be all for that as well.